pandas/core/ops.py

```diff
@@ -622,6 +622,10 @@ def _is_offset(self, arr_or_obj):
         """ check if obj or all elements of list-like is DateOffset """
         if isinstance(arr_or_obj, ABCDateOffset):
             return True
+        elif (is_datetime64_dtype(arr_or_obj) or
+              is_timedelta64_dtype(arr_or_obj)):
```

Comment:
catch period here as well?

Reply:
What binary ops is a `PeriodIndex` valid in?

Reply:
``is_period``

Reply:
sorry, we have ``is_period_dtype`` which is the correct to use here